containers/azure-terraform/.vscode/tasks.json

```diff
@@ -0,0 +1,107 @@
+{
+    "version": "2.0.0",
+    "tasks": [
+        {
+            "label": "az login",
+            "type": "shell",
+            "command": "az login && az account set -s ${env:ARM_SUBSCRIPTION_ID}",
+            "problemMatcher": []
+        },
+        {
+            "label": "terraform create backend",
+            "type": "shell",
+            "linux": {
+                "command": "${workspaceFolder}/test-project/create-backend.sh"
+            },
+            "problemMatcher": []
+        },
+        {
+            "label": "terraform init",
+            "type": "shell",
+            "command": "terraform",
+            "problemMatcher": [],
+            "args": [
+                "init",
+                "-backend-config=\"storage_account_name=${env:TF_BACKEND_STORAGE_ACCOUNT}\"",
+                "-backend-config=\"container_name=${env:TF_BACKEND_CONTAINER}\"",
+                "-backend-config=\"key=${env:TF_BACKEND_KEY}\"",
+                "-backend-config=\"resource_group_name=${env:TF_BACKEND_RESOURCE_GROUP}\"",                              
+                "-backend-config=\"subscription_id=${env:ARM_SUBSCRIPTION_ID}\"",
+                "-backend-config=\"tenant_id=${env:ARM_TENANT_ID}\"",
+            ],
+            "options": {
+                "cwd": "${workspaceFolder}/test-project/templates"
```

Comment:
Not sure about this tasks:

* Azure: Sign In command is available in the container.
* Rest of commands are already accessible in the command palette. as `Azure Terraform: ...`
* If we add `"azureTerraform.terminal": "integrated"` to the settings all the commands will be available to run by default inside the container instead of cloud shell --> Set as default?
* If samples terraform files are copied to the working folder and user executes `Azure Terraform: init` the plugin will ask for the backend parameters.
* Azure Terraform plugin commands run in the current folder so if user wants to test the provided sample should open the terminal, then `cd test-project/templates/` and select the desired command from the control palette.

Reply:
Thank you for the feedback! We have removed the `AzureTerraform` extension from the dev container to avoid any confusion as these task configurations are separate. 

The current configurations also include tasks to `create backend`, `lint`, `format`, and the `az login` task also sets the default subscription.

Based on your feedback, we will look into contributing back to the Azure Terraform Extension. Thank you for all your insights on this. 

Please let us know if this will work or if you would recommend anything else

Reply:
Cool we are almost there!

The developer container name is `azure-terraform` so it is the natural home for the `azure terraform` extension and that is the reason behind installing `graphviz` in the container: https://github.com/microsoft/vscode-dev-containers/blob/b81d9b51743d686935a30ece9f059d787e5207cc/containers/azure-terraform/.devcontainer/Dockerfile#L17 

I think that the extension should stay in place and instead just keep the tasks that make sense such as `create backend`, `lint` and `format` delegating the rest of actions to the extension commands which work on any folder by default.

Reply:
Thank you so much @cmendible. We agree with the recommendation. We have adjusted the README and removed the tasks that are similarly configured through the extension (validate, plan, apply, destroy).  

We kept `az login`, as `Azure:Sign in` didn't work as expected in the devcontainer and `terraform init` for better developer experience so that the user can leverage the task configuration without having to enter the backend parameters every time. 

As mentioned above we can definitely look into contributing to the extension.

Reply:
Great just added some comments so you can please fix a format issue inside the README and remove the sample related var from the definition.

Reply:
Thank you for the feedback again, we have made the adjustments.